client/components/site-selector/index.jsx

```diff
@@ -185,7 +185,7 @@ const SiteSelector = React.createClass( {
 					href={ this.props.siteBasePath ? siteHref : null }
 					key={ 'site-' + site.ID }
 					indicator={ this.props.indicator }
-					onSelect={ this.onSiteSelect.bind( this, site.slug ) }
+					onSelect={ this.onSiteSelect.bind( this, site.slug, site.ID ) }
```

Comment:
To get the constructor we should re-write this component using ES6 class and on the fly we should remove the mixin to finally implement its behavior in another way.